### PR TITLE
Added integration tests documentation

### PR DESCRIPTION
**Pull request by Marketionist**
**This pull request adds integration tests documentation**

Also:
- typos fixed for ```unit_tests.rst``` and ```integration_tests.rst```
- updated links in ```general.rst``` and ```index.rst```
- ```gulpfile.js``` updated with new path to integration tests
- added dummy page object ```page.dummy.js```

### CODE REVIEW

File: docs/testing/general.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ interactions.
 This testing infrastructure includes them both with `Jasmine
 <http://jasmine.github.io/>`_ as the test suite and `Karma
 <http://karma-runner.github.io/>`_ as the test runner for unit tests.
-`Protractor <protractortest.org>`_ serves as the integration tests framework.
+`Protractor <http://angular.github.io/protractor>`_ serves as the integration tests framework.
```
Comment (FinalAngel) [inline docs/testing/general.rst]: why not ``protractortest.org`` which is a more generic domain? WE'd have to add ``http://``

File: docs/testing/integration_tests.rst [hunk 2]
```diff
@@ -2,4 +2,50 @@
...
-- To be completed
+
+Configuration
+=============
+
+The main configuration file to look at is ``/tests/protractor.conf.js``.
+It configures our **specs**, **exclude** and **suites** paths.
+
+Inside **specs** we specify the location of all specs that should be launched.
+It can be set to run all tests that are in the specs folder like this ``specs: ['specs/*.js']``.
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: you would have to refere to `tests/integration/specs`

File: docs/testing/integration_tests.rst [hunk 3]
```diff
@@ -2,4 +2,50 @@
...
+
+The **exclude** setting allows us to specifically exclude certain specs from
+running. By defining a path - say ``specs/spec.log*.js`` in ``exclude`` - we
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: same here

File: docs/testing/integration_tests.rst (hunk 2 above)
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: in general our specs live in `integration` or?

File: docs/testing/integration_tests.rst [hunk 4]
```diff
@@ -2,4 +2,50 @@
...
+from being launched.
+
+Alternatively, **suites** may be used - we can group spec files into suites
+with the keyword. For example we can add 2 tests into suite like this:
+
+.. code-block:: javascript
+
+    // Spec patterns are relative to the location of the spec file. They may
+    // include glob patterns.
+    suites: {
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: so what should be the file organisation?

```
/tests/integration/
  - spec.article.create.js
  - spec.article.delete.js
  - spec.login.js
  - spec.logout.js
```

and than the rest is managed through configuration?

File: docs/testing/integration_tests.rst [hunk 5]
```diff
@@ -2,4 +2,50 @@
...
+
+Or you can also specify specs using patterns:
+
+.. code-block:: javascript
+
+    suites: {
+        logInOut: ['specs/spec.log*.js']
+    }
+
+You can find more information about this within the
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: the specs are currently define within gulp: https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/develop/gulpfile.js#L147 it loads all spec files

File: docs/testing/integration_tests.rst (hunk 2 above)
Comment (Marketionist) [inline docs/testing/integration_tests.rst]: Right, will fix

File: docs/testing/general.rst (hunk 1 above)
Comment (Marketionist) [inline docs/testing/general.rst]: Sounds good, will be updated

File: docs/testing/integration_tests.rst (hunk 4 above)
Comment (Marketionist) [inline docs/testing/integration_tests.rst]: It should be:

```
/tests/integration/specs
  - spec.article.create.js
  - spec.article.delete.js
  - spec.login.js
  - spec.logout.js

/tests/integration/pages
  - page.article.create.js
  - page.article.delete.js
  - page.login.js
  - page.logout.js
```

Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: what's the difference between `specs` and `pages`?

File: docs/testing/index.rst [hunk 6]
```diff
@@ -6,7 +6,7 @@ Testing
 
     This section describes the unit and integration tests setup using
     `Jasmine <http://jasmine.github.io/>`_ and
-    `Protractor <protractortest.org>`_. You will find advise on how to setup
+    `Protractor <http://www.protractortest.org>`_. You will find advise on how to setup
```
Comment (evildmp) [inline docs/testing/index.rst]: advise > advice

File: docs/testing/integration_tests.rst [hunk 7]
```diff
@@ -2,4 +2,51 @@
...
 
-- To be completed
+
+Configuration
+=============
+
+The main configuration file to look at is ``/tests/protractor.conf.js``.
+It configures our **specs**, **exclude** and **suites** paths.
+
+Inside **specs** we specify the location of all specs that should be launched.
```
Comment (evildmp) [inline docs/testing/integration_tests.rst]: Inside > In

File: docs/testing/integration_tests.rst [hunk 8]
```diff
@@ -2,4 +2,51 @@
...
+
+Or you can also specify specs using patterns:
+
+.. code-block:: javascript
+
+    suites: {
+        logInOut: ['integration/specs/spec.log*.js']
+    }
+
+You can find more information about this within the
```
Comment (evildmp) [inline docs/testing/integration_tests.rst]: within > in

Generally, "in" is more appropriate than "inside" or "within".

File: tests/integration/pages/page.dummy.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+//#############################################################################
+'use strict';
+/* jshint browser: true */
+/* jshint shadow: true */
+
+/**
+ * This file uses the Page Object pattern to define a test.
+ **/
+
+var DummyPage = {
```
Comment (FinalAngel) [inline tests/integration/pages/page.dummy.js]: should that be capitalized? /cc @vxsx

Comment (vxsx) [inline tests/integration/pages/page.dummy.js]: > Name variablesLikeThis, ClassesLikeThis, CONSTANTS_LIKE_THIS and events-like-this.

from http://aldryn-boilerplate-bootstrap3.readthedocs.org/en/latest/guidelines/javascript.html#naming
doesn't look like a class to me, so nope

File: docs/testing/integration_tests.rst [hunk 10]
```diff
@@ -2,4 +2,49 @@
...
+=============
+
+The main configuration file to look at is ``/tests/protractor.conf.js``.
+It configures our **browserName**.
+
+In **browserName** we specify the browser that will be used to launch the tests.
+It can be set to ``phantomjs``, ``firefox`` or ``chrome``.
+
+You can find more information about this in the
+`protractor referenceConf.js <https://github.com/angular/protractor/blob/master/docs/referenceConf.js>`_ documentation.
```
Comment (evildmp) [inline docs/testing/integration_tests.rst]: Can we keep docs to 100 columns please?

File: tests/integration/pages/page.dummy.js (hunk 9 above)
Comment (Marketionist) [inline tests/integration/pages/page.dummy.js]: It's a variable to keep an object inside, so renamed it to dummyPage

File: docs/testing/integration_tests.rst (hunk 10 above)
Comment (Marketionist) [inline docs/testing/integration_tests.rst]: Yes, sure. Fixed

File: docs/testing/integration_tests.rst [hunk 11]
```diff
@@ -2,4 +2,50 @@
 Integration Tests
 *****************
 
-- To be completed
+
+Configuration
+=============
+
+The main configuration file to look at is ``/tests/protractor.conf.js``.
+It configures our **browserName**.
```
Comment (evildmp) [inline docs/testing/integration_tests.rst]: Just spotted this - if "browserName" is a literal (e.g. a variable name) it should be in double-backticks - ``browserName``. See http://docs.django-cms.org/en/develop/contributing/contributing.html#inline-markup.

Comment (Marketionist) [inline docs/testing/integration_tests.rst]: Tnx for noting, fixed

**Comment by vxsx** [comment]
LGTM, but I would appreciate another opinion as well

File: docs/testing/integration_tests.rst [hunk 12]
```diff
@@ -2,4 +2,50 @@
...
+You can find more information about this in the
+`protractor referenceConf.js
+<https://github.com/angular/protractor/blob/master/docs/referenceConf.js>`_ documentation.
+
+All spec files should be placed in ``/tests/integration/specs`` and all page
+object files should be in ``/tests/integration/pages``. So, the file organisation
+structure is:
+
+```
+/tests/integration/specs
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: use tree structure similar to https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/master/docs/structure/general.rst#private

**Comment by FinalAngel** [comment]
LGTM except for the tree issue

File: docs/testing/integration_tests.rst (hunk 12 above)
Comment (Marketionist) [inline docs/testing/integration_tests.rst]: Ok, done

File: docs/testing/integration_tests.rst [hunk 13]
```diff
@@ -2,4 +2,51 @@
...
+`protractor referenceConf.js
+<https://github.com/angular/protractor/blob/master/docs/referenceConf.js>`_ documentation.
+
+All spec files should be placed in ``/tests/integration/specs`` and all page
+object files should be in ``/tests/integration/pages``. So, the file organisation
+structure is:
+
+.. code-block:: text
+
+    tests/
```
Comment (FinalAngel) [inline docs/testing/integration_tests.rst]: nice job

**Pull request merged**